cmd/kubeadm/app/cmd/init.go

```diff
@@ -196,7 +196,7 @@ func (i *Init) Run(out io.Writer) error {
 	// TODO this is not great, but there is only one address we can use here
 	// so we'll pick the first one, there is much of chance to have an empty
 	// slice by the time this gets called
-	masterEndpoint := fmt.Sprintf("https://%s:%d", i.cfg.API.AdvertiseAddresses[0], i.cfg.API.Port)
+	masterEndpoint := fmt.Sprintf("https://%s:%d", i.cfg.API.BindAddress, i.cfg.API.BindPort)
```

Comment:
In the defaulting, `0.0.0.0` which is a placeholder, is replaced for the real address of the default net interface